src/Storefront/Controller/RegisterController.php

```diff
@@ -230,7 +230,7 @@ private function getAdditionalRegisterValidationDefinitions(DataBag $data, Sales
             ]));
         }
 
-        if ($this->systemConfigService->get('core.loginRegistration.requirePasswordConfirmation', $context->getSalesChannel()->getId())) {
+        if ($data->get('guest') !== '1' && $this->systemConfigService->get('core.loginRegistration.requirePasswordConfirmation', $context->getSalesChannel()->getId())) {
```

Comment:
```suggestion
        if ($data->has('guest') && $this->systemConfigService->get('core.loginRegistration.requirePasswordConfirmation', $context->getSalesChannel()->getId())) {
```

like in the other code above :wink:

Reply:
I think that should be `! $data->has('guest')` right ? Since we are checking user is not guest ?

Reply:
of course 😃👍

Reply:
It is done :)